vllm/engine/llm_engine.py

```diff
@@ -333,9 +346,12 @@ def _init_tokenizer(self, **tokenizer_init_kwargs):
             trust_remote_code=self.model_config.trust_remote_code,
             revision=self.model_config.tokenizer_revision)
         init_kwargs.update(tokenizer_init_kwargs)
+
```

Comment:
> I see - why did we need to define the new generator function instead of leaving it as is? Ideally we shouldn't make changes unless they're necessary.

I wasn't able to overload async generators properly, so I had to wrap the generator inside the function. I'll try to apply the same pattern as `LLM._run_engine` to avoid the need for overloads.